src-self-hosted/codegen/c.zig

```diff
@@ -41,16 +40,33 @@ fn renderType(file: *C, writer: std.ArrayList(u8).Writer, T: Type, src: usize) !
     }
 }
 
+fn renderValue(file: *C, writer: std.ArrayList(u8).Writer, val: Value, src: usize) !void {
+    switch (val.tag()) {
```

Comment:
It's OK to switch on the tag for a "hot path" but really, this should be switching on the `zigTypeTag()` and dealing with the value on a per-type basis. When you write skeleton code like this, it's good to keep in mind the contributors that will look at this code after you and try to improve it. As is, the structure here is misleading to contributors. I suggest to switch on the type, and check if it's an unsigned integer 64 bits or less, and then call `toUnsignedInt`.